src/network/mcpe/handler/InGamePacketHandler.php

```diff
@@ -371,6 +372,8 @@ private function handleUseItemTransaction(UseItemTransactionData $data) : bool{
 			case UseItemTransactionData::ACTION_CLICK_AIR:
 				if($this->player->isUsingItem()){
 					if(!$this->player->consumeHeldItem()){
+						$hungerAttr = $this->player->getAttributeMap()->get(Attribute::HUNGER) ?? throw new AssumptionFailedError();
+						$this->session->syncAttributes($this->player, [$hungerAttr]);
```

Comment:
`markSynchronized(false)` would suit better, so that it gets synced together with other attributes at the end of the tick.

Reply:
Done